### DISCUSSION TITLE
CDK to make templates, without managing deploy?

### DISCUSSION THREAD

**Discussion by VTLee** [aws/aws-cdk#16910] [Q&A]
We use Cloudformation heavily in our business. We're planning to change the process with our "next generation", and we'd like something with better customization options than CF alone allows through conditional statements... It seems right up the alley of CDK...

Except CDK is only ever discussed as a deployment tool from a command-line perspective.. We'd ideally like to do something like:

1. Something drops a message to SQS
2. SQS triggers a lambda event, which builds the template in CDK based on inputs from the message
3. Lambda fires off CDK/CF stack creation and exits

We currently do this exact process entirely in a lambda, but it has to select from predefined CF templates with appropriate conditionals and it passes in the overrides necessary... but we really want to swap to dynamic CF templates so we can have more complex situations without making new CF Templates for each combination of features.

Can the CDK be leveraged in this way?
Would it continue to be supported? (with the obvious limitations that we lose some of the help CDK provides)

It looks like we can use CDK to make the template, but the only option is to write to disk... Which should be fine since we would likely need to upload the file to S3 for CloudFormation to reference. I'm willing to go that route, but I don't want to make a system that is really against the CDK's design, and is likely to break in the near future.

Thanks for your thoughts.

-Lee

**Comment by peterwoodworth** [comment]
CDK is incredible at creating CICD, it's one of the biggest reasons to use CDK! Infrastructure that meets your requirements could be configured entirely through Pipelines https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html
